### ISSUE TITLE
How to enable both mask and keypoint in your config ?

### ISSUE THREAD

**Issue by adamadatasci** [facebookresearch/detectron2#1219]
## ❓How to enable both mask and keypoint in your config ?

I want to train both keypoint and instance segmentation using MaskRcnn or CascadRcnn:

Here is my config:

-------------------------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------------------------
1 - Data Registration -- working fine

from detectron2.data import DatasetCatalog, MetadataCatalog

DatasetCatalog.register("my_dataset7", get_data)

MetadataCatalog.get("my_dataset7").set(thing_classes=["F","G","I","E"]).set(keypoint_names = ['w1',"w2"]).set(keypoint_connection_rules = [('w1','w2',(0,255,0))]).set(keypoint_flip_map=[('w1','w2')]).set(thing_colors = [(0, 0, 255),(255, 0, 0),(255, 0, 255),(0, 255, 255)])

metadata = MetadataCatalog.get("my_dataset7")

-------------------------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------------------------
2 - Training -- No Error, but there is no Keypoint value in the output

cfg = get_cfg()

cfg.merge_from_file("./detectron2_repo/configs/COCO-InstanceSegmentation/mask_rcnn_X_101_32x8d_FPN_3x.yaml")

cfg.DATASETS.TRAIN = ("my_dataset7",)
cfg.DATASETS.TEST = ()
cfg.DATALOADER.NUM_WORKERS = 2

cfg.MODEL.WEIGHTS = "detectron2://COCO-InstanceSegmentation/mask_rcnn_X_101_32x8d_FPN_3x/139653917/model_final_2d9806.pkl" 

cfg.SOLVER.IMS_PER_BATCH = 2
cfg.SOLVER.BASE_LR = 0.00025
cfg.SOLVER.MAX_ITER = 1000   
cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 128

cfg.MODEL.ROI_HEADS.NUM_CLASSES = 4

cfg.MODEL.ROI_KEYPOINT_HEAD.NUM_KEYPOINTS = 2

cfg.MODEL.MASK_ON = True
cfg.MODEL.KEYPOINT_ON = True

--------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------
3- Reference:
https://detectron2.readthedocs.io/modules/config.html#config-references

**Comment by ppwwyyxx** [comment]
As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.

We provide configs & models with standard academic settings and expect users to have the knowledge to choose or design appropriate models & parameters for their own tasks.

**Comment by adamadatasci** [comment]
It would be interesting to show an example of building a model from scratch using Detectron2 in colab. The documentation is hard to follow, lot of uncommented code and their role.

**Comment by adamadatasci** [comment]
342         self.mask_on      = cfg.MODEL.MASK_ON
    343         # fmt: on
--> 344         assert not cfg.MODEL.KEYPOINT_ON
    345 
    346         self.pooler = ROIPooler(

AssertionError: 


So it's not possible with Detectron2. You've said in another thread that it was possible.

mutil task training #17
https://github.com/facebookresearch/detectron2/issues/17

**Comment by abrichr** [comment]
The following runs without any error:

```
config = 'detectron2/configs/COCO-Keypoints/keypoint_rcnn_R_50_FPN_3x.yaml'
opts = [
  'MODEL.WEIGHTS',
  model_zoo.get_checkpoint_url("COCO-Keypoints/keypoint_rcnn_R_50_FPN_3x.yaml"),
  'MODEL.MASK_ON', True,
  'MODEL.SEM_SEG_HEAD.LOSS_WEIGHT', 0.5
]
cfg = get_cfg()
cfg.merge_from_file(config)
cfg.merge_from_list(opts)
predictor = DefaultPredictor(cfg)
predictions = predictor(frame)
```

However the resulting inferred masks are just noise, since the model weights loaded from the specified checkpoint haven't been trained:

![image](https://user-images.githubusercontent.com/774615/80855874-e8210700-8c12-11ea-92e4-df9ca619171a.png)

Hopefully this should work if the model is retrained, but that's expensive. Otherwise you can simply run inference twice per frame, once using `keypoint_rcnn_R_50_FPN_3x.yaml` and again using `panoptic_fpn_R_101_3x.yaml`, but this reduces throughput by half.

Ideally it would be possible to extract the SEM_SEG_HEAD weights from a model trained with `panoptic_fpn_R_101_3x.yaml` and combine them with the weights from a model trained with `keypoint_rcnn_R_50_FPN_3x.yaml`.  @ppwwyyxx is this possible?

**Comment by 17sarf** [comment]
Hi, this is not directly related to this topic, but  I am trying to use the keypoint RCNN to generate skeletons for detected objects. I have managed to get the keypoints as seen in the picture below, however, I'm not sure how to connect them. I am unsure how to set the `keypoint_names` and `keypoint_connection_rules` in my `MetadataCatalog`. 

![image](https://user-images.githubusercontent.com/44234385/85846102-3c70df00-b79d-11ea-81a2-15cdca2cc71e.png)
